annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastFieldResolver.java

```diff
@@ -0,0 +1,60 @@
+package com.exactpro.epfast.annotation.processing;
+
+import com.exactpro.epfast.annotations.FastField;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import java.util.*;
+import java.util.stream.Collectors;
+
+class FastFieldResolver {
+    private HashMap<Element, FastTypeElement> cache = new HashMap<>();
+
+    private Elements elements;
+
+    public FastFieldResolver(Elements elements) {
+        this.elements = elements;
+    }
+
+    private boolean isOverriddenIn(FastTypeElement fastType, FastFieldElement fastField) {
+        return fastType.getFastFields().stream()
+            .anyMatch(field -> field.getMethodName().equals(fastField.getMethodName()) &&
+                field.getParameterType().equals(fastField.getParameterType()));
+    }
+
+    private List<FastFieldElement> getFastFieldsFor(Element aClass) {
+        if (cache.containsKey(aClass)) {
+            return cache.get(aClass).getFastFields();
+        }
+        return aClass.getEnclosedElements().stream()
+            .filter(element -> element.getAnnotation(FastField.class) != null)
+            .map(FastFieldElement::new)
+            .collect(Collectors.toList());
+    }
+
+    private Element getSuperClass(Element element) {
+        TypeMirror superClass = ((TypeElement) element).getSuperclass();
+        return elements.getTypeElement(superClass.toString());
+    }
+
+    public void resolveFields(FastTypeElement fastType) {
+        if (cache.containsKey(fastType.getElement())) {
```

Comment:
Lasha, we need to cache result of `getFastFieldsFor` method. The way you did it caches only classes with `@FastType` annotation. I find it very unlikely that a type with `@FastType` will be extended from another annotated class. More viable scenario is that base class is't annotated.